src/citrine/resources/status_detail.py

```diff
@@ -0,0 +1,35 @@
+from typing import TypeVar, Union
+
+from citrine._serialization.serializable import Serializable
+from citrine._serialization import properties
+
+from gemd.enumeration.base_enumeration import BaseEnumeration
+
+
+class StatusLevelEnum(BaseEnumeration):
+    """The level of the status message."""
+
+    INFO = "Info"
+    WARNING = "Warning"
+    ERROR = "Error"
+    UNKOWN = "Unknown"
+
+
+Self = TypeVar('Self', bound='StatusDetail')
+
+
+class StatusDetail(Serializable[Self]):
```

Comment:
I don't see this `Self` pattern anywhere else in the library. Does it have a risk of namespace collision? Not super familiar with the implementation details of this chunk of our serialization stack. Closest matches look like they would call this the `StatusDetailType`
https://github.com/CitrineInformatics/citrine-python/blob/13e8eb1f123222863903f627ea82442e59c224a1/src/citrine/resources/module.py#L37

```suggestion
StatusDetailType = TypeVar('StatusDetailType', bound='StatusDetail')


class StatusDetail(Serializable[StatusDetailType]):
```

Reply:
I copied it from src/citrine/_rest/resource.py, but I think your way makes more sense.